ui/Modal/Search.svelte

```diff
@@ -0,0 +1,171 @@
+<script lang="typescript">
+  import { push } from "svelte-spa-router";
+  import { createEventDispatcher } from "svelte";
+
+  import * as notification from "../src/notification";
+  import * as path from "../src/path";
+  import type { Project } from "../src/project";
+  import * as remote from "../src/remote";
+  import {
+    projectRequest as request,
+    projectSearch as store,
+    reset,
+    requestProject,
+    searchProject,
+    urnValidationStore,
+  } from "../src/search";
+  import { ValidationStatus } from "../src/validation";
+
+  import { Icon, Input } from "../DesignSystem/Primitive";
+  import { Remote, TrackToggle } from "../DesignSystem/Component";
+
+  let id: string;
+  let value: string;
+
+  const dispatch = createEventDispatcher();
+  const urnValidation = urnValidationStore();
+
+  const navigateToProject = (project: Project) => {
+    dispatch("hide");
+    push(path.projectSource(project.id));
+  };
+  const navigateToUntracked = () => {
+    dispatch("hide");
+    push(path.projectUntracked(value));
+  };
+  const onKeydown = (event: KeyboardEvent) => {
+    switch (event.code) {
+      case "Enter":
+        // Navigate to project directly if present.
+        if ($store.status === remote.Status.Success) {
+          // FIXME(xla): Once remote/Remote offer stronger type guarantees this needs to go.
+          navigateToProject(
+            ($store as { status: remote.Status.Success; data: Project }).data
+          );
+        }
+        break;
+      case "Escape":
+        dispatch("hide");
+        break;
+    }
+  };
+  const onTrack = () => {
+    requestProject({ urn: value });
+  };
+
+  // Validate input entered, at the moment valid RadUrns are the only acceptable input.
+  $: if (value && value.length > 0) {
+    urnValidation.validate(value);
+    id = value.replace("rad:git:", "");
+  } else {
+    urnValidation.reset();
+  }
+  // To support quick pasting, request the urn once valid to get tracking information.
+  $: if ($urnValidation.status === ValidationStatus.Success) {
+    searchProject({ urn: value });
+  }
+  // Reset searches if the input became invalid.
+  $: if ($urnValidation.status !== ValidationStatus.Success) {
+    reset();
+  }
+  // Fire notification when a request has been created.
+  $: if ($request.status === remote.Status.Success) {
+    notification.info(
+      "You’ll be notified on your profile when this project has been found.",
+      false,
+      "View profile",
+      () => {
+        dispatch("hide");
+        push(path.profileProjects());
+      }
+    );
+  }
+
+  $: tracked = $store.status === remote.Status.Success;
+  $: untracked = $store.status === remote.Status.Error;
+</script>
+
+<style>
+  .container {
+    width: 26.5rem;
+  }
+
+  .search-bar {
+    margin-bottom: 1rem;
+    position: relative;
+  }
```

Comment:
```suggestion
  .container {
    width: 50vw;
    max-width: 800px;
    min-width: 500px;
  }

  .search-bar {
    margin-bottom: 1rem;
    position: relative;
    padding: var(--content-padding);
    background-color: var(--color-background);
    border-radius: 4px;
  }
```

Now the validation error message is displaying in red on top of the overlay which is dark, looking no good, imo:

 
![x](https://user-images.githubusercontent.com/7075260/95957171-a3bfa700-0dff-11eb-8984-7f444d3e021e.png)


This suggestion does two things:

1. Improves the responsiveness of the modal container, looking better in both small and large screens:

![small](https://user-images.githubusercontent.com/7075260/95957244-c18d0c00-0dff-11eb-93d7-7a61c8877e9e.png)
![medium](https://user-images.githubusercontent.com/7075260/95957256-c651c000-0dff-11eb-8835-296da8eba367.png)

![Screenshot from 2020-10-14 09-25-07](https://user-images.githubusercontent.com/7075260/95957317-db2e5380-0dff-11eb-96c9-9b26e3606ed6.png)

2. Wraps the search bar with the appropriate background colour with nice padding:

![Screenshot from 2020-10-14 09-32-38](https://user-images.githubusercontent.com/7075260/95957632-38c2a000-0e00-11eb-8f77-6aa122b351dd.png)

Reply:
Thanks for the suggestion, @radicle-dev/product is aware of it and the work is tracked in #1031

Reply:
Woudln't hurt to include this alrerady though as an temporary improvement.

Reply:
We can make these changes when polishing for the beta release. Don't see much value in doing changes until @radicle-dev/product has settled on something.